src/typing_extensions.py

```diff
@@ -935,6 +995,8 @@ def __new__(cls, name, bases, ns, *, total=True, closed=False):
                 if type(base) is not _TypedDictMeta and base is not typing.Generic:
                     raise TypeError('cannot inherit from both a TypedDict type '
                                     'and a non-TypedDict base class')
+            if closed is not None and extra_items is not NoExtraItems:
+                raise TypeError(f"Cannot combine closed={closed!r} and extra_items")
```

Comment:
I think this branch may also be uncovered, but feel free to merge and tackle this as a followup!